src/addrs.ts

```diff
@@ -1,40 +1,67 @@
 import { networkInterfaces, NetworkInterfaceInfo } from 'os'
-import { nodeToMultiaddr } from './utils'
+import { isLinkLocaleAddress, nodeToMultiaddr } from './utils'
 import { Multiaddr } from 'multiaddr'
 import Debug from 'debug'
 const log = Debug('hopr-connect')
 
 import { isLocalhost, ipToU8aAddress, isPrivateAddress } from './utils'
 
+type AddrOptions = {
+  interface?: string
+  useIPv4?: boolean
+  useIPv6?: boolean
+  includePrivateIPv4?: boolean
+  includeLocalhostIPv4?: boolean
+  includeLocalhostIPv6?: boolean
+}
+
+function validateOptions(opts?: AddrOptions) {
```

Comment:
why `?`

Reply:
It supports also getting any address when passing no opts object, but this functionality is currently not used. But we could remove it

Reply:
changed